src/App.tsx

```diff
@@ -15,48 +18,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort(
+    (prevGood, nextGood) => {
+      switch (sortType) {
+        case SortType.LENGTH:
+          return prevGood.length - nextGood.length;
+        case SortType.ALPHABET:
+          return prevGood.localeCompare(nextGood);
+        default:
+          return 0;
+      }
+    },
+  );
+
+  return !isReversed
+    ? visibleGoods
+    : visibleGoods.reverse();
+}
+
+export const App = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState<SortType>(SortType.NONE);
+
+  const options = {
+    sortType,
+    isReversed,
+  };
+
+  const handleSort = (sortByProp: SortType) => () => setSortType(sortByProp);
+
+  const handleSortAlphabeticaly = handleSort(SortType.ALPHABET);
+
+  const handleSortByLength = handleSort(SortType.LENGTH);
+
+  const handleReverse = () => setIsReversed((previousValue) => !previousValue);
+
+  const handleReset = () => {
+    setSortType(SortType.NONE);
+    setIsReversed(false);
+  };
+
+  const reordered = isReversed || sortType !== SortType.NONE;
+
+  const goodsToRender = getReorderedGoods(
+    goodsFromServer,
+    options,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={
+            classNames('button', 'is-info',
+              { 'is-light': sortType !== SortType.ALPHABET })
+          }
+          onClick={handleSortAlphabeticaly}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames(
+            'button is-success',
+            { 'is-light': sortType !== SortType.LENGTH },
+          )}
+          onClick={handleSortByLength}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
```

Comment:
Use multiline formatting for better readability when using `classNames` function. 📖